Add a plan file to allow for internal Apple testing of a new tentative MotionMark branch

smfr opened the pull request:
#### 4383ea973361df52eeae10483203a5902952b8a7
<pre>
Add a plan file to allow for internal Apple testing of a new tentative MotionMark branch
<a href="https://bugs.webkit.org/show_bug.cgi?id=261450">https://bugs.webkit.org/show_bug.cgi?id=261450</a>
rdar://115327200

Reviewed by Stephanie Lewis.

Add a plan file and relevant patches so that Apple can test
<a href="https://github.com/WebKit/MotionMark/tree/apple-tentative-1.3">https://github.com/WebKit/MotionMark/tree/apple-tentative-1.3</a> internally.

Plan file was optimized via plan_file_optimizer.py.

* Tools/Scripts/webkitpy/benchmark_runner/benchmark_builder.py:
(BenchmarkBuilder._apply_patch):
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/signposts/MotionMark1.3-tentative.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/webdriver/MotionMark1.3-tentative.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/webserver/MotionMark1.3-tentative.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3-tentative.plan: Added.

Canonical link: <a href="https://commits.webkit.org/267904@main">https://commits.webkit.org/267904@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f0739c88b55547af678425d0214202022e929b70

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/18020 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/18348 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/18914 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/19854 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/16869 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/18217 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/21644 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/18506 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/18860 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/18237 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/18495 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/15680 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/20731 "Built successfully") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/18172 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/15722 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/16433 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/22967 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/16738 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/16603 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/20835 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/17169 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/14557 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/16267 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/4287 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/20628 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/17019 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

dewei-zhu commented on Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3-tentative.plan:
```diff
@@ -0,0 +1,252 @@
+    },
+    "output_file": "motionmark.result",
+    "github_subtree": {
+        "url": "https://api.github.com/repos/webkit/MotionMark/tree/e15bbf1e075007a3e20d1bd56a97b80b812c48b8/",
```
Should be `https://api.github.com/repos/WebKit/MotionMark/git/trees/db6987eb65f3be06ad409c50fe9807845e2e2123`.

dewei-zhu commented on Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3-tentative.plan:
```diff
@@ -0,0 +1,252 @@
+            {"path": "tests/core", "mode": "040000", "type": "tree"},
+            {"path": "tests/core/canvas-stage.html", "mode": "100644", "type": "blob"},
+            {"path": "tests/core/design.html", "mode": "100644", "type": "blob"},
+            {"path": "tests/core/focus.html", "mode": "100644", "type": "blob"},
```
This was removed. Per discussion, `github_subtree` needs to be updated accordingly.
The way to get this updated accordingly is:
1. Remove existing `github_subtree` in `Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3-tentative.plan`
2. Run `python3 Tools/Scripts/webkitpy/benchmark_runner/plan_file_optimizer.py Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3-tentative.plan`

smfr committed:
Add a plan file to allow for internal Apple testing of a new tentative MotionMark branch
https://bugs.webkit.org/show_bug.cgi?id=261450
rdar://115327200

Reviewed by Stephanie Lewis.

Add a plan file and relevant patches so that Apple can test
https://github.com/WebKit/MotionMark/tree/apple-tentative-1.3 internally.

Plan file was optimized via plan_file_optimizer.py.

* Tools/Scripts/webkitpy/benchmark_runner/benchmark_builder.py:
(BenchmarkBuilder._apply_patch):
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/signposts/MotionMark1.3-tentative.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/webdriver/MotionMark1.3-tentative.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/patches/webserver/MotionMark1.3-tentative.patch: Added.
* Tools/Scripts/webkitpy/benchmark_runner/data/plans/motionmark1.3-tentative.plan: Added.

Canonical link: https://commits.webkit.org/267904@main
